src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -560,49 +577,53 @@ public void updateLastCheckin(Consumer consumer) {
 
     @Transactional
     public void updateLastCheckin(Consumer consumer, Date checkinDate) {
-        String hql = "UPDATE Consumer c SET c.lastCheckin = :date, c.updated = :date WHERE c.id = :cid";
+        if (consumer == null || checkinDate == null) {
+            throw new IllegalArgumentException();
```

Comment:
deleted and noted

Reply:
For clarity, the consumer check is a valid once since we'll NPE if it's null otherwise, and I'm not opposed to both checks existing; just that if both are going to exist, they should be separate checks.